test/hotspot/gtest/utilities/test_bitMap_truncate.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "precompiled.hpp"
+#include "logging/logStream.hpp"
+#include "memory/arena.hpp"
+#include "memory/resourceArea.hpp"
+#include "utilities/bitMap.inline.hpp"
+#include "unittest.hpp"
+
+class BitMapTruncateTest {
+
+ public:
+  const static BitMap::idx_t BITMAP_SIZE = 128;
+
+  template <class ResizableBitMapClass>
+  static void fillBitMap(ResizableBitMapClass& map,  BitMap::idx_t size) {
+    BitMap::idx_t set_bits[] = {0, 31, 63, 64, 95, 127};
+    for (BitMap::idx_t bit : set_bits) {
+        if (bit < size) {
+            map.set_bit(bit);
+        }
+    }
+  }
+
+  template <class ResizableBitMapClass>
+  static void testTruncate(BitMap::idx_t start_bit, BitMap::idx_t end_bit, ResizableBitMapClass& result) {
+    ResourceMark rm;
+
+    ResizableBitMapClass map(BITMAP_SIZE);
+    fillBitMap(map, BITMAP_SIZE);
+    map.truncate(start_bit, end_bit);
+
+    EXPECT_TRUE(map.is_same(result));
+  }
+
+  template <class ResizableBitMapClass>
+  static void testTruncateOneWord() {
+    ResourceMark rm;
+
+    ResizableBitMapClass map(64);
+    map.set_bit(0);
+    map.set_bit(1);
+    map.set_bit(2);
+    map.set_bit(3);
+
+    ResizableBitMapClass result(2);
+    result.set_bit(0);
+    result.set_bit(1);
+
+    map.truncate(1, 3);
+
+    EXPECT_TRUE(map.is_same(result));
+  }
+
+  template <class ResizableBitMapClass>
+  static void testTruncateSame() {
+    // Resulting map should be the same as the original
+    ResourceMark rm;
+    ResizableBitMapClass map(BITMAP_SIZE);
+    fillBitMap(map, BITMAP_SIZE);
+    testTruncate<ResizableBitMapClass>(0, BITMAP_SIZE, map);
+  }
```

Comment:
`testTruncateOneWord()` is pretty easy to understand, but I found it really hard to follow the logic of the other test cases. For example, it took me a while to understand the variable `map` in `testTruncateSame()` is not the same as the `map` in `testTruncate()`.

Could all the tests be rewritten to the same style of `testTruncateOneWord()`, where the entire logic is inside the same function?

Reply:
Also, I think the contents of this file should be moved inside test_bitMap.cpp, so you don't need to repeat things like TestArenaBitMap.